keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -711,214 +647,324 @@ BucketAccess {
-    |      }                                      |       |      }                                        |
-    | }                                           |       |   }                                           |
-    |---------------------------------------------|       | }                                             |
-                                                          |-----------------------------------------------|
+
+The returned `bucketID` should be a unique identifier for the OSP bucket known to the driver. This value will be used by COSI to make all subsequent calls related to this bucket, so the driver must be able to correlate `bucketID` to the OSP backend bucket. It is easiest for drivers to use the request `existingBucketID` field the `bucketID`, but this is not strictly required.
+
+##### DriverGrantBucketAccess
+
+This gRPC call creates a set of access credentials for a bucket. This API must be idempotent.
+
+`bucketID` used for input is the same ID returned by the driver in [DriverCreateBucket](#drivercreatebucket) (dynamically-provisioned) or [DriverGetExistingBucket](#drivergetexistingbucket) (statically-provisioned).
+
+This `accountName` field is the concatenation of the characters `ba-` (short for BucketAccess) and the BucketAccess UID. It is used as the idempotency key for requests to the drivers regarding a particular BA. The driver should ensure that multiple DriverGrantBucketAccess calls for the same `accountName` do not result in more than one OSP backend bucket access being provisioned corresponding to that name. Using or appending random identifiers can lead to multiple unused bucket accesses being created in the OSP backend in the event of timing-related driver/sidecar failures or restarts.
+
+Input `accessParameters` are the opaque parameters copied from the BucketAccessClass. Drivers can use these parameters to configure OSP bucket access features based on the Administrator's BucketAccessClass configuration.
+
+The returned `accountID` should be a unique identifier for the account in the OSP. This value will be included in all subsequent calls to the driver for changes to the BucketAccess. As `bucketID` is to Bucket, `accountID` is to BucketAccess.
+
+Returned `credentials` will be transformed into [BucketAccess secret data](#bucketaccess-secret-data).
+
+```go
+DriverGrantBucketAccess(DriverGrantBucketAccessRequest) DriverGrantBucketAccessResponse
+
+DriverGrantBucketAccessRequest{
+  "bucketID": "<Bucket.status.bucketID>", // e.g., "ba-<BucketClaim.UID>", "my-static-bucket"
+  "accountName": "ba-<BucketAccess.UID>"
+  "protocol": "<protocol>", // e.g., "S3", copied from BucketAccess.spec.protocol
+  "parameters": { // copied from BucketAccess.status.parameters
+      "key": "value",
+      // ...
+  }
+}
+
+DriverGrantBucketAccessResponse {
+  "accountID": "<ID returned by driver>", // will be applied to BucketAccess.status.accountID
+  "credentials": { // credentials info (authn/authx)
```

Comment:
As in the comment above. When empty, sidecar should treat this as a failure, even when no error was returned.